source/binding/Statements.cpp

```diff
@@ -528,10 +534,23 @@ bool StatementList::verifyConstantImpl(EvalContext& context) const {
     return true;
 }
 
+void StatementList::serializeTo(ASTSerializer& serializer) const {
+    serializer.startArray("list");
+    for (auto const& stmt : list) {
+        serializer.serialize(*stmt);
+    }
+    serializer.endArray();
+}
+
 BlockStatement::BlockStatement(const StatementBlockSymbol& block, SourceRange sourceRange) :
     Statement(StatementKind::Block, sourceRange), blockKind(block.blockKind), block(&block) {
 }
 
+void BlockStatement::serializeTo(ASTSerializer& serializer) const {
+    serializer.write("blockKind", toString(blockKind));
+    serializer.write("list", *list);
```

Comment:
Only one of the block or list members will be non-null, so you need to check which one here. Instead, I would recommend calling this "body" and setting it to getStatements() which will do that check for you.

The reason for this split is because some block statements in Verilog create a new level of hierarchy, in which case there will be a BlockSymbol here, and some do not, in which case this will just be a list of statements.